ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -364,6 +367,35 @@ private void cleanTablesFromBootstrap() throws HiveException, IOException, Inval
     }
   }
 
+  /**
+   * If replication policy is changed between previous and current load, then the excluded tables in
+   * the new replication policy will be dropped.
+   * @throws HiveException Failed to get/drop the tables.
+   */
+  private void dropTablesExcludedInReplScope(ReplScope replScope) throws HiveException {
+    // If all tables are included in replication scope, then nothing to be dropped.
+    if ((replScope == null) || replScope.includeAllTables()) {
+      return;
+    }
+
+    Hive db = getHive();
+    String dbName = replScope.getDbName();
+
+    // List all the tables that are excluded in the current repl scope.
+    Iterable<String> tableNames = Collections2.filter(db.getAllTables(dbName),
+        tableName -> {
+          assert(tableName != null);
+          return !tableName.toLowerCase().startsWith(
```

Comment:
why prepare the list ..can we not just drop the table ?

Reply:
We need to get all tables right? Otherwise, how do we know which table to drop?

Reply:
I got your point... But this is Iterable. We list from getAllTables and then the below loop iterate and just drop when found a match. So, it is ideally the same as what you expect. Also, not sure if we can handle exceptions in filter method. Anyways, this is same as we don't have any double loops.